src/actions/actions.ts

```diff
@@ -67,4 +69,21 @@ export class Actions {
 
     return response;
   }
+
+  async constructAction({
+    payload,
+    sigHeader,
+    secret,
+    tolerance = 300,
```

Comment:
@faroceann calling out that I reduced this because it seems like actions should be timely... can you see any reason this would be a mistake?

Reply:
Good call out, this should be safe